broadcast-message.php

```diff
@@ -47,23 +52,38 @@
    }
    else
    {
+      $wanted_domains = array_intersect($allowed_domains, $_POST['domains']);
+
       $table_mailbox = table_by_key('mailbox');
-	  $table_alias = table_by_key('alias');
-      
-	  $q = "select username from $table_mailbox union select goto from $table_alias " .
-		   "where goto not in (select username from $table_mailbox)";
-
-      $result = db_query ($q);
-      if ($result['rows'] > 0)
-      {
+      $table_alias = table_by_key('alias');
+
+      $recipients = [];
+
+      foreach( $wanted_domains as $domain) {
+         $q = "SELECT username from $table_mailbox WHERE domain = '$domain'";
+         if(!isset($_POST['mailboxes_only']) || $_POST['mailboxes_only'] != "on") {
+            $q .= " UNION SELECT GOTO FROM $table_alias WHERE domain = '$domain' " .
+                  " AND goto NOT IN ($q)";
```

Comment:
No problem, I've been busy too...

I've changed the `foreach`-Loop to `db_in_clause` and it seems to work (I've just tested it once on my server...)

The refactoring to move this code to the Mailbox-/AliasHandler is a to complicated job to do it during exams, maybe I'll have a look at this when I'm less busy then now